src/components/search-by-drug-orders/search-by-drug-orders.resource.ts

```diff
@@ -0,0 +1,50 @@
+import { openmrsFetch } from "@openmrs/esm-framework";
+import useSWRImmutable from "swr/immutable";
+
+import { DropdownValue, Response } from "../../types";
+
+/**
+ * @returns Drugs
+ */
+export function useDrugs() {
+  const { data, error } = useSWRImmutable<{
+    data: { results: Response[] };
+  }>("/ws/rest/v1/drug", openmrsFetch);
+
+  const drugs: DropdownValue[] = [];
+  data?.data.results.map((drug: Response, index: number) => {
+    drugs.push({
+      id: index,
+      label: drug.display,
+      value: drug.uuid,
+    });
+  });
+  return {
```

Comment:
Even if you are using the immutable, it will not change the data, but the returned value will be **a new object** with the **same old values**.

Reply:
but the docs say that once the data is cached, they will never request it again. If it never requests why it returns a new object?

https://swr.vercel.app/docs/revalidation#disable-automatic-revalidations

Reply:
Yes, I agree that the **SWR's data object** doesn't change, because this is a cached data, and whenever this hook is rendered, the `data` object will not change.

The concern I'm talking about is the object you are returning, made on Line 22, this particular object gets recreated everytime this hook is called, given that the **SWR's** data object is a cached unchanged data. It's like creating new object again and again, with the same data in place. 

That's why useMemo comes in here. It doesn't create a new object on every call of this hook, it sends the same object with the same address.

@brandones, please correct if I'm wrong.
Thanks!

Reply:
@vasharma05 is correct. @anjula-sack this is about the difference between **reference equality** and **value equality.** Even if the values are the same, you are instantiating a new object with `{}` (think of it as doing something like `new Object()`. If you don't have a clear understanding of the two types of equality and how they work in JavaScript and React you should read up on the topic; it is one of those computer science fundamentals that is actually very important to know.

Reply:
Got it, Thanks for the explanation @brandones, @vasharma05